src/main/java/uk/gov/hmcts/reform/mi/miextractionservice/component/impl/CoreCaseDataExportBlobDataComponentImpl.java

```diff
@@ -0,0 +1,152 @@
+
+        List<OutputCoreCaseData> outputData = new ArrayList<>();
+
+        List<String> blobNameIndexes = getYearMonthIndexes(fromDate, toDate);
+
+        for (BlobContainerItem blobContainerItem : sourceBlobServiceClient.listBlobContainers()) {
+            if (blobContainerItem.getName().startsWith(CCD_DATA_CONTAINER_PREFIX)) {
+                for (BlobItem blobItem : sourceBlobServiceClient.getBlobContainerClient(blobContainerItem.getName()).listBlobs()) {
+                    if (blobNameIndexes.parallelStream().anyMatch(blobItem.getName()::contains)) {
+
+                        byte[] blobData = blobDownloadComponent
+                            .downloadBlob(sourceBlobServiceClient, blobContainerItem.getName(), blobItem.getName());
+
+                        List<String> stringData = readerUtil.readBytesAsStrings(blobData);
+
+                        outputData.addAll(convertDataStringsToOutput(stringData, fromDate, toDate));
+                    }
+                }
+            }
+        }
+
+        if (Boolean.TRUE.equals(outputData.isEmpty())) {
+            throw new ExportException("No data to output.");
+        }
+
+        csvWriterComponent.writeBeansAsCsvFile(WORKING_FILE_NAME, outputData);
+
+        encryptArchiveComponent.createEncryptedArchive(Collections.singletonList(WORKING_FILE_NAME), WORKING_ARCHIVE);
+
+        BlobContainerClient blobContainerClient = targetBlobServiceClient.getBlobContainerClient(CCD_OUTPUT_CONTAINER_NAME);
+
+        if (!blobContainerClient.exists()) {
+            blobContainerClient.create();
+        }
+
+        String outputBlobName = fromDate.format(dateTimeUtil.getDateFormat())
+            + NAME_DELIMITER
+            + toDate.format(dateTimeUtil.getDateFormat())
+            + NAME_DELIMITER
+            + WORKING_ARCHIVE;
+
+        blobContainerClient.getBlobClient(outputBlobName).uploadFromFile(WORKING_ARCHIVE, true);
+
+        return generateBlobUrlComponent.generateUrlForBlob(targetBlobServiceClient, CCD_OUTPUT_CONTAINER_NAME, outputBlobName);
+    }
+
+    private List<OutputCoreCaseData> convertDataStringsToOutput(List<String> data, OffsetDateTime fromDate, OffsetDateTime toDate) {
+        List<CoreCaseData> coreCaseDataList = new ArrayList<>();
+        data.forEach(string -> coreCaseDataList.add(dataParserComponent.parse(string)));
+
+        List<OutputCoreCaseData> outputCoreCaseDataList = new ArrayList<>();
+        coreCaseDataList.forEach(coreCaseData -> {
+            OffsetDateTime eventCreatedDate = OffsetDateTime.ofInstant(Instant.ofEpochMilli(coreCaseData.getCeCreatedDate()), ZoneOffset.UTC);
+            if (eventCreatedDate.isAfter(fromDate) && eventCreatedDate.isBefore(toDate)) {
+                outputCoreCaseDataList.add(coreCaseDataFormatterComponent.formatData(coreCaseData));
+            }
+        });
+
+        return outputCoreCaseDataList;
+    }
+
+    private List<String> getYearMonthIndexes(OffsetDateTime fromDate, OffsetDateTime toDate) {
```

Comment:
Just an alternative option :smiley:

```suggestion
    private List<String> getYearMonthIndexes(OffsetDateTime fromDate, OffsetDateTime toDate) {
          DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM");
        List<String> listOfIndexes = new ArrayList<>();
        OffsetDateTime current = fromDate.withDayOfMonth(1);
        while(current.isBefore(toDate)) {
            listOfIndexes.add(DATE_TIME_FORMATTER.format(current));
            current = current.plusMonths(1);
        }
        return listOfIndexes;
        }
```